### PR TITLE
Add baseUrl option (closes #1932)

### LINKED ISSUE #1932: Global Start URL Option
### Are you requesting a feature or reporting a bug?
Request a feature

### What is the current behavior?
Start Urls must be specified on fixtures or tests

### What is the requested behavior?
It would be nice if I could set a global start url, instead of doing it for every fixture.

My app is a single page app, so for all tests, my start url is just localhost:PORT, and then I just navigate to the page I am testing with my app. It would be nice to only specify the start url in one spot on either the cmd line or the test runner (if using the node api).

I think it could be the lowest priority, so it would be overridden by a fixture.page call, which would then be overridden by test.page

### PR DESCRIPTION
**Pull request by Artem-Babich**
## Purpose
In some cases, it is necessary to specify the common URL for all tests. Also it is necessary to have an ability to specify the base URL for all the relative paths specified in tests.

## Approach
Add a base-url CLI arg and baseUrl config property and pass this arg to all the compilers. Initialize this arg in the Test/Fixture object on a page method call.

## References
[#1932](https://github.com/DevExpress/testcafe/issues/1932)

## Pre-Merge TODO
- [x] Write tests for your proposed changes
- [x] Make sure that existing tests do not fail

### CODE REVIEW

**Review by AlexKamaev — APPROVED** [review]

**Review by titerman — APPROVED** [review]

File: src/api/exportable-lib/index.js [hunk 1]
```diff
@@ -73,10 +73,10 @@ const exportableLib = {
     },
```
Comment (AndreyBelym) [inline src/api/exportable-lib/index.js:73]: Do not merge until docs are ready.

File: src/cli/argument-parser.ts [hunk 2]
```diff
@@ -173,6 +173,7 @@ export default class CLIArgumentParser {
             .option('--disable-multiple-windows', 'disable multiple windows mode')
             .option('--disable-http2', 'disable the HTTP/2 proxy backend and force the proxy to use only HTTP/1.1 requests')
             .option('--cache', 'cache web assets between test runs')
+            .option('--baseUrl <url>', 'set the base url for all tests')
```
Comment (Aleksey28) [inline src/cli/argument-parser.ts:176]: You should use `kebab-case` for naming CLI options. It automatically converted to `camelCase` for the config.

**Review by Aleksey28 — CHANGES_REQUESTED** [review]
Please, process the comment and rebase

**Review by Aleksey28 — APPROVED** [review]

**Comment by Nisden** [comment]
Is there a reason for not supporting baseUrl when using Role's?

**Comment by Artem-Babich** [comment]
[https://github.com/DevExpress/testcafe/pull/6894](https://github.com/DevExpress/testcafe/pull/6894)

Hello @Nisden,

Supporting relative URLs in a Role is a separate task, like supporting relative URLs in any other functionality that allows you to use some URL. That is why currently the baseURL option does not work with roles. We have a separate GitHub issue regarding this enhancement: [Support relative urls in Roles](https://github.com/DevExpress/testcafe/issues/3406). You can leave your feedback there. We will be happy to implement this functionality in TestCafe after we collect enough user votes.

**Pull request merged**